src/sorting/heap_sort.rs

```diff
@@ -122,4 +122,32 @@ mod tests {
         heap_sort(&mut arr);
         assert_eq!(&arr, &[1, 2, 3, 4]);
     }
+
+    #[test]
+    fn odd_number_of_elements() {
+        let mut arr = vec![3, 4, 2, 1, 7];
+        heap_sort(&mut arr);
+        assert_eq!(&arr, &[1, 2, 3, 4, 7]);
+    }
+
+    #[test]
+    fn unusual_element_at_end() {
+        let mut arr = vec![3, 4, 2, 1, 7, 33333];
+        heap_sort(&mut arr);
+        assert_eq!(&arr, &[1, 2, 3, 4, 7, 33333]);
+    }
+
+    #[test]
+    fn unusual_element_at_start() {
```

Comment:
Do we need these unusual tests?

Reply:
@AnshulMalik I don't know if we _need_ them, but I've learned that when testing, unusual end cases should definitely be considered.

Reply:
However, if you feel that these tests are unnecessary, I can remove them.

Reply:
Yep, feels like redundant to me. Better if we keep them concise :)